crates/indexer/src/http/metadata_json.rs

```diff
@@ -528,3 +546,31 @@ pub async fn process<'a>(
 
     Ok(())
 }
+
+async fn dispatch_metadata_document(
+    client: &Client,
+    addr: String,
+    image: &str,
+    raw: Value,
+) -> Result<()> {
+    let image = Url::parse(image)
+        .ok()
+        .and_then(|u| {
+            let id = AssetIdentifier::new(&u);
+
+            proxy_url(client.proxy_args(), &id, Some(("width", "400")))
+                .map(|o| o.map(|u| u.to_string()))
+                .transpose()
+        })
+        .unwrap_or_else(|| Ok(image.to_owned()))?;
+
+    let document = MetadataDocument { image, raw };
```

Comment:
Is this enough data to create a link to the NFT? Wouldn't we need metadata_address?

Reply:
That's in the `id` field of the document we dispatch for upsert.